.github/workflows/discover-lean-pr-testing.yml

```
@@ -179,36 +179,38 @@ jobs:
         SUCCESSFUL_MERGES="${{ steps.execute-merges.outputs.successful_merges }}"
         FAILED_MERGES="${{ steps.execute-merges.outputs.failed_merges }}"
 
-        {
-          printf $'msg<<EOF\n'
-
-          # Report successful merges
-          if [ -n "$SUCCESSFUL_MERGES" ]; then
-            printf $'### ✅ Successfully merged branches:\n\n'
-            for MERGE_INFO in $SUCCESSFUL_MERGES; do
-              IFS='|' read -r PR_NUMBER GITHUB_DIFF BRANCH <<< "$MERGE_INFO"
-              printf $'- lean#%s ([diff](%s))\n' "$PR_NUMBER" "$GITHUB_DIFF"
-            done
-            printf $'\n'
-          else
-            printf $'No branches were successfully merged.\n\n'
-          fi
+        # Start building the message
+        MESSAGE=""
+
+        # Report successful merges
+        if [ -n "$SUCCESSFUL_MERGES" ]; then
+          MESSAGE+=$'### ✅ Successfully merged branches:\n\n'
+          for MERGE_INFO in $SUCCESSFUL_MERGES; do
+            IFS='|' read -r PR_NUMBER GITHUB_DIFF BRANCH <<< "$MERGE_INFO"
+            MESSAGE+=$(printf '- lean#%s ([diff](%s))\n' "$PR_NUMBER" "$GITHUB_DIFF")
```

```suggestion
            MESSAGE+=$(printf -- '- lean#%s ([diff](%s))\n' "$PR_NUMBER" "$GITHUB_DIFF")
```
Tells `printf` that the following `-` is not a stumped flag.